ts/packages/node/package.json

```diff
@@ -51,6 +51,7 @@
     "typeorm": "^0.3.16"
   },
   "devDependencies": {
-    "@types/lodash": "^4.14.195"
+    "@types/lodash": "^4.14.195",
+    "keyv": "^4.5.2"
```

Comment:
shouldn't `keyv` be under "dependencies" prop?

Reply:
I've been thinking about this, and keyv could be a dependency of the project. But I'm not sure if the keyv backends could be. Once they are numerous, (counting only the official ones, there are 9), we would be adding a lot of dependencies to the project.

Based on my understanding of the documentation, - "You can choose the most suitable ConfigStore based on your specific needs and workflow" - users of ConfigStores will generally use only one implementation. 

For example, someone who only uses the AWS Secrets Manager ConfigStore is unlikely to use any of the other implementations. Therefore, there is no need to have mysql, mssql, pg, axios, etc. as dependencies of the project. If they need another implementation in addition to the AWS Secrets Manager, let's say Postgres, they would download the Configu package with only the Postgres implementation, to use alongside the Configu core package.

Of course, what I'm suggesting here would be a drastic change, and it needs to be discussed and studied to see if it makes sense for the project as a whole.

But what I propose for now is trying to do something similar to this in the `KeyvConfigStore`: we could add only the Keyv as a dependency of the project, and if the client wants to use a custom Keyv Backend, they would have to install the adapter in addition to the Configu package. 

That would be something similar to what keyv suggests in their documentation:

>Usage
Install Keyv.
```bash
npm install --save keyv
```
>By default everything is stored in memory, you can optionally also install a storage adapter.
```bash
npm install --save @keyv/redis
npm install --save @keyv/etcd
```
>Create a new Keyv instance, passing your connection string if applicable. Keyv will automatically load the correct storage adapter.

```ts
const Keyv = require('keyv');

// One of the following
const keyv = new Keyv();
const keyv = new Keyv('redis://user:pass@localhost:6379');
const keyv = new Keyv('etcd://localhost:2379');
```

Reply:
@gabrielcipriano,
You've brought up a compelling perspective. Balancing flexibility with simplicity is always a delicate act, and the idea of modularizing ConfigStores is intriguing.

I see the merits in your suggestion. By allowing users to install only the specific stores they need, we could reduce unnecessary bloat in their projects. However, this could become challenging with SDKs in languages where package management isn't as flexible as npm. Additionally, there's the consideration that a significant portion of users might leverage multiple stores concurrently.

I believe the best approach might be to engage the broader community in this discussion. Your idea deserves wider attention, so I propose opening a dedicated issue for it and inviting feedback. Concurrently, I'll investigate potential solutions that allow us to package each language SDK without mandating all dependencies. I'll link the issue back here once it's live.

Thank you for prompting this discussion—it's insights like these that help refine and improve our project!

cc @configu/dev

Reply:
For now, before merging it Im suggesting that we will move `keyv` to the `dependencies` section